ibis/backends/datafusion/compiler.py

```diff
@@ -937,34 +938,30 @@ def extract_hour(op, **kw):
 
 @translate.register(ops.ExtractMillisecond)
 def extract_millisecond(op, **kw):
-    def ms(array: pa.Array) -> pa.Array:
-        return pc.cast(pc.millisecond(array), pa.int32())
-
-    extract_milliseconds_udf = df.udf(
-        ms,
-        input_types=[PyArrowType.from_ibis(op.arg.dtype)],
-        return_type=PyArrowType.from_ibis(op.dtype),
-        volatility="immutable",
-        name="extract_milliseconds_udf",
-    )
     arg = translate(op.arg, **kw)
-    return extract_milliseconds_udf(arg)
+
+    if op.arg.dtype.is_date():
+        return registry.UDFS["extract_millisecond_time"](arg)
```

Comment:
It looks like we may not have test coverage for some of these. Would you mind adding some in a follow-up PR?